### PR TITLE
Cleanup and PgUpdate support

### PR DESCRIPTION
**Pull request by EmadAlblueshi**

### CODE REVIEW

File: src/main/java/com/julienviet/pgclient/impl/PreparedUpdate.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+package com.julienviet.pgclient.impl;
+
+import com.julienviet.pgclient.PgUpdate;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.ext.sql.UpdateResult;
+
+public class PreparedUpdate implements PgUpdate {
```
Comment (vietj) [inline src/main/java/com/julienviet/pgclient/impl/PreparedUpdate.java]: rather `PgUpdateImpl`

**Comment by vietj** [comment]
overall we should not duplicate the same class with a params and a param-less version. Param-less should be params command with empty params list.

**Comment by EmadAlblueshi** [comment]
Done  https://github.com/vietj/vertx-pg-client/issues/14 https://github.com/vietj/vertx-pg-client/issues/15 and https://github.com/vietj/vertx-pg-client/issues/18

File: src/main/java/com/julienviet/pgclient/impl/PreparedBatchCommand.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.julienviet.pgclient.codec.util.Util.*;
+
+/**
+ * @author <a href="mailto:emad.albloushi@gmail.com">Emad Alblueshi</a>
+ */
+
+class PreparedBatchCommand extends UpdateCommandBase {
```
Comment (vietj) [inline src/main/java/com/julienviet/pgclient/impl/PreparedBatchCommand.java]: please keep PreparedUpdateCommand and consider that an update is a batch of a single unit.

Comment (EmadAlblueshi) [reply src/main/java/com/julienviet/pgclient/impl/PreparedBatchCommand.java]: Yes now we have only PreparedUpdateCommand and can be used for batch and update

File: src/test/java/com/julienviet/pgclient/PgConnectionTestBase.java [hunk 3]
```diff
@@ -1060,6 +920,44 @@ public void testSetAndGetSerializableTx(TestContext ctx) {
     });
   }
 
+  @Test
+  public void testPgUpdate(TestContext ctx) {
```
Comment (vietj) [inline src/test/java/com/julienviet/pgclient/PgConnectionTestBase.java]: just testUpdate no need to say PgUpdate here

File: src/test/java/com/julienviet/pgclient/PgConnectionTestBase.java [hunk 4]
```diff
@@ -1060,6 +920,44 @@ public void testSetAndGetSerializableTx(TestContext ctx) {
...
+          .execute(ctx.asyncAssertSuccess(r -> {
+            ctx.assertEquals("PgClient Rocks!", r.getRows().get(0).getValue("message"));
+            async.complete();
+          }));
+      }));
+    }));
+  }
+
+  @Test
+  public void testPgUpdateWithParams(TestContext ctx) {
```
Comment (vietj) [inline src/test/java/com/julienviet/pgclient/PgConnectionTestBase.java]: testUpdateWithParams

**Comment by vietj** [comment]
The documentation is not covering the update case, it should be updated (after query and before batch).

File: src/test/java/com/julienviet/pgclient/PgConnectionTestBase.java (hunk 3 above)
Comment (EmadAlblueshi) [reply src/test/java/com/julienviet/pgclient/PgConnectionTestBase.java]: The testUpdate will conflict with other test case so it's better to rename to testPreparedUpdate

File: src/test/java/com/julienviet/pgclient/PgConnectionTestBase.java (hunk 4 above)
Comment (EmadAlblueshi) [reply src/test/java/com/julienviet/pgclient/PgConnectionTestBase.java]: This will be renamed to testPreparedUpdateWithParams

**Pull request merged**